##### Excercise. Create a program that asks the user to input a number n, and finds the sum of the given series. Also, check if the sum is likely to converge or not.


###### 
$ \frac{1}{1!} + \frac{2}{2!} +  ...  + \frac{n}{n!} $

In [2]:
def sum(n):
    s = 0
    f = 1
    for i in range(1, n+1):
        for j in range(1, i+1):
            f = f*j
        s += i/f
    print("The required sum is:", s)
    
while True:
    n = input("Enter n: ")
    if n == "":
        break
    sum(int(n))

Enter n: 2
The required sum is: 2.0
Enter n: 5
The required sum is: 2.264033564814815
Enter n: 8
The required sum is: 2.264033805997176
Enter n: 33
The required sum is: 2.264033805997176
Enter n: 56
The required sum is: 2.264033805997176
Enter n: 


Therefore, this sum is convergent at 2.264033805997176

###### 
$ \frac{1}{2} + \frac{3}{4} +  ...  + \frac{n-1}{n} $

In [8]:
while True:
    n = input("\nEnter n (even): ")
    if n == "":
        break
    n = int(n)
    sum = 0.0
    for i in range(1, n+1, 2):
        sum += i/(i+1)
    print(f"The sum of the series for n = {n} is {sum}.")


Enter n (even): 4
The sum of the series for n = 4 is 1.25.

Enter n (even): 8
The sum of the series for n = 8 is 2.9583333333333335.

Enter n (even): 22
The sum of the series for n = 22 is 9.490061327561328.

Enter n (even): 56
The sum of the series for n = 56 is 26.036414480516818.

Enter n (even): 88
The sum of the series for n = 88 is 41.813637053335455.

Enter n (even): 222
The sum of the series for n = 222 is 108.35437819637353.

Enter n (even): 874
The sum of the series for n = 874 is 433.6708537058073.

Enter n (even): 1000
The sum of the series for n = 1000 is 496.60358828500523.

Enter n (even): 2000
The sum of the series for n = 2000 is 996.2572645697257.

Enter n (even): 7000
The sum of the series for n = 7000 is 3495.6310616186333.

Enter n (even): 


Therefore, this sum is not likely to converge.

###### 
$ 1 + (1+2) + (1+2+3) + ... + (1+2+3+...+n) $

In [9]:
while True:
    n = input("\nEnter n: ")
    if n == "":
        break
    n = int(n)
    sum = 0.0
    for i in range(1, n+1):
        for j in range(1, i+1):
            sum += j
    print(f"The sum of the series for n = {n} is {sum}.")


Enter n: 4
The sum of the series for n = 4 is 20.0.

Enter n: 7
The sum of the series for n = 7 is 84.0.

Enter n: 67
The sum of the series for n = 67 is 52394.0.

Enter n: 87
The sum of the series for n = 87 is 113564.0.

Enter n: 


Therefore, this sum will not converge.

###### 
$ \frac{1}{2!} + \frac{3}{4!} +  ...  + \frac{n-1}{n!} $

In [11]:
from math import factorial

while True:
    n = input("\nEnter n (even): ")
    if n == "":
        break
    n = int(n)
    sum = 0.0
    for i in range(1, n+1, 2):
        sum += i/factorial((i+1))
    print(f"The sum of the series for n = {n} is {sum}.")


Enter n (even): 4
The sum of the series for n = 4 is 0.625.

Enter n (even): 8
The sum of the series for n = 8 is 0.6321180555555556.

Enter n (even): 22
The sum of the series for n = 22 is 0.6321205588285577.

Enter n (even): 66
The sum of the series for n = 66 is 0.6321205588285577.

Enter n (even): 


Therefore, this sum is convergent at 0.6321205588285577.

###### 
$ \frac{1!}{x} - \frac{3!}{x^2} + \frac{5!}{x^3} - \space ... \space + (up \space to \space n \space terms) $

In [16]:
from math import factorial, pow

while True:
    n = input("\nEnter n (number of terms): ")
    if n == "":
        break
    n = int(n)
    x = int(input("\nEnter x: "))
    sum = 0.0
    a, b = list(), list()
    for i in range(1, 2*n, 2):
        a.append(factorial(i))
    for j in range(1, n+1):
        b.append(pow(x, j))
    for k in range(len(a)):
        if k%2 == 0:
            sum += a[k]/b[k]
        else:
            sum -= a[k]/b[k]
    print(f"The sum of the series for {n} terms is {sum}.")


Enter n (number of terms): 3

Enter x: 8
The sum of the series for 3 terms is 0.265625.

Enter n (number of terms): 9

Enter x: 8
The sum of the series for 9 terms is 2574961.4979248047.

Enter n (number of terms): 6

Enter x: 8
The sum of the series for 6 terms is -142.1611328125.

Enter n (number of terms): 5

Enter x: 7
The sum of the series for 5 terms is 19.86214077467722.

Enter n (number of terms): 2

Enter x: 12
The sum of the series for 2 terms is 0.041666666666666664.

Enter n (number of terms): 12

Enter x: 2
The sum of the series for 12 terms is -6.286698879435987e+18.

Enter n (number of terms): 


Therefore, this sum is not likely to converge.

###### Pi Series
$ 4 \space (1 - \frac{1}{3} + \frac{1}{5} - \frac{1}{7} + \frac{1}{9} \space ... \space + (up \space to \space n \space terms)) $

In [22]:
while True:
    n = input("\nEnter n (number of terms) (grater than 1): ")
    if n == "":
        break
    n = int(n)
    l = list()
    sum_ = 1.0
    for i in range(3, (2*n), 2):
        l.append(i)
    for j in range(len(l)):
        if j%2 == 0:
            sum_ -= 1/l[j]
        else:
            sum_ += 1/l[j]
    sum = 4*sum_
    print(f"The sum of the pi series for {n} terms is {sum}.")


Enter n (number of terms) (grater than 1): 30
The sum of the pi series for 30 terms is 3.108268566698947.

Enter n (number of terms) (grater than 1): 50
The sum of the pi series for 50 terms is 3.121594652591011.

Enter n (number of terms) (grater than 1): 80
The sum of the pi series for 80 terms is 3.129093141775721.

Enter n (number of terms) (grater than 1): 125
The sum of the pi series for 125 terms is 3.1495925256000317.

Enter n (number of terms) (grater than 1): 200
The sum of the pi series for 200 terms is 3.136592684838816.

Enter n (number of terms) (grater than 1): 400
The sum of the pi series for 400 terms is 3.1390926574960143.

Enter n (number of terms) (grater than 1): 900
The sum of the pi series for 900 terms is 3.140481542821618.

Enter n (number of terms) (grater than 1): 2000
The sum of the pi series for 2000 terms is 3.1410926536210413.

Enter n (number of terms) (grater than 1): 4000
The sum of the pi series for 4000 terms is 3.1413426535937043.

Enter n (number 

Therefore, the pi series does converge to the known value of pi, i.e., 3.1415.